examples/step-9/step-9.cc

```diff
@@ -800,58 +712,51 @@ namespace Step9
                 {
                   for (unsigned int j = 0; j < dofs_per_cell; ++j)
                     copy_data.cell_matrix(i, j) -=
-                      (face_advection_directions[q_point] *
+                      (scratch_data.face_advection_directions[q_point] *
                        scratch_data.fe_face_values.normal_vector(q_point) *
                        scratch_data.fe_face_values.shape_value(i, q_point) *
                        scratch_data.fe_face_values.shape_value(j, q_point) *
                        scratch_data.fe_face_values.JxW(q_point));
 
                   copy_data.cell_rhs(i) -=
-                    (face_advection_directions[q_point] *
+                    (scratch_data.face_advection_directions[q_point] *
                      scratch_data.fe_face_values.normal_vector(q_point) *
-                     face_boundary_values[q_point] *
+                     scratch_data.face_boundary_values[q_point] *
                      scratch_data.fe_face_values.shape_value(i, q_point) *
                      scratch_data.fe_face_values.JxW(q_point));
                 }
         }
 
-
-    // Now go on by transferring the local contributions to the system of
-    // equations into the global objects. The first step was to obtain the
-    // global indices of the degrees of freedom on this cell.
+    // The final piece of information the copy routine needs is the cell DoF
```

Comment:
I am not sure if using the abbreviation `DoF` is a good idea here. I would prefer `are the global indices of the degrees of freedom on this cell`.

Reply:
Fixed